packages/playwright-core/src/server/injected/reactSelectorEngine.ts

```diff
@@ -144,9 +144,16 @@ function findReactRoots(root: Document | ShadowRoot, roots: ReactVNode[] = []):
   const walker = document.createTreeWalker(root, NodeFilter.SHOW_ELEMENT);
   do {
     const node = walker.currentNode;
+    // ReactDOM Legacy client API:
     // @see https://github.com/baruchvlz/resq/blob/5c15a5e04d3f7174087248f5a158c3d6dcc1ec72/src/utils.js#L329
-    if (node.hasOwnProperty('_reactRootContainer'))
+    if (node.hasOwnProperty('_reactRootContainer')) {
       roots.push((node as any)._reactRootContainer._internalRoot.current);
+    } else {
+      // React 17+
+      const rootKey = Object.keys(node).find(key => key.startsWith('__reactContainer'));
```

Comment:
React does set it when mounting and it has a random suffix, see here: https://github.com/facebook/react/blob/a724a3b578dce77d427bef313102a4d0e978d9b4/packages/react-dom/src/client/ReactDOMComponentTree.js#L62-L64

Reply:
This is an awesome bit of information! Can we have this as an inline comment in code?